internal/terraform/node_resource_plan_instance.go

```diff
@@ -778,7 +778,7 @@ func (n *NodePlannableResourceInstance) importState(ctx EvalContext, addr addrs.
 	}
 
 	importedState := states.NewResourceInstanceObjectFromIR(imported[0])
-	if deferred == nil && importedState.Value.IsNull() {
+	if deferred == nil && importType == "ID" && importedState.Value.IsNull() {
```

Comment:
This appears to already exist, but it should probably be turned into a pre-defined enum rather than an ad-hoc string at some point